stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/attachments/content/MediaAttachmentContent.kt

```diff
@@ -0,0 +1,485 @@
+    val isVideo = attachment.type == AttachmentType.VIDEO
+
+    // Used as a workaround for Coil's lack of a retry policy.
+    // See: https://github.com/coil-kt/coil/issues/884#issuecomment-975932886
+    var retryHash by remember {
+        mutableStateOf(0)
+    }
+
+    val data =
+        if (isImage || (isVideo && ChatTheme.videoThumbnailsEnabled)) {
+            attachment.imagePreviewUrl
+        } else {
+            null
+        }
+
+    val painter = rememberStreamImagePainter(
+        model = ImageRequest.Builder(LocalContext.current)
+            .data(data)
+            .setParameter(key = "retry_hash", value = retryHash)
+            .build()
+    )
+
+    val mixedMediaPreviewLauncher = rememberLauncherForActivityResult(
+        contract = MediaGalleryPreviewContract(),
+        onResult = { result -> onMediaGalleryPreviewResult(result) }
+    )
+
+    // Used to refresh the request for the current page
+    // if it has previously failed.
+    if (data != null && connectionState == ConnectionState.CONNECTED &&
+        painter.state is AsyncImagePainter.State.Error
+    ) {
+        retryHash++
+    }
+
+    val areVideosEnabled = ChatTheme.videoThumbnailsEnabled
+
+    Box(
+        modifier = modifier
+            .background(Color.Black)
+            .fillMaxWidth()
+            .combinedClickable(
+                interactionSource = MutableInteractionSource(),
+                indication = rememberRipple(),
+                onClick = {
+                    mixedMediaPreviewLauncher.launch(
+                        MediaGalleryPreviewContract.Input(
+                            message = message,
+                            initialPosition = attachmentPosition,
+                            videoThumbnailsEnabled = areVideosEnabled
+                        )
+                    )
+                },
+                onLongClick = { onLongItemClick(message) }
+            ),
+        contentAlignment = Alignment.Center
+    ) {
+        val backgroundColor =
+            if (isImage) ChatTheme.colors.imageBackgroundMessageList
+            else ChatTheme.colors.videoBackgroundMessageList
+
+        Image(
+            modifier = modifier
+                .fillMaxSize()
+                .background(backgroundColor),
+            painter = painter,
+            contentDescription = null,
+            contentScale = ContentScale.Crop
+        )
+
+        MediaPreviewPlaceHolder(
+            asyncImagePainterState = painter.state,
+            progressIndicatorStrokeWidth = 3.dp,
+            progressIndicatorFillMaxSizePercentage = 0.25f,
+            isImage = isImage,
+            placeholderIconTintColor = ChatTheme.colors.disabled
+        )
+
+        if (isVideo) {
+            playButton()
+        }
+    }
+}
+
+/**
+ * A simple play button that is overlaid above
+ * video attachments.
+ *
+ * @param modifier The modifier used for styling.
+ */
+@Composable
+public fun PlayButton(
```

Comment:
This doesn't have to be public, given that it's easy to C/P and most peopel won't reuse it really